### DISCUSSION TITLE
Improved progress and feedback UX/UI

### DISCUSSION THREAD

**Discussion by ruben-arts** [prefix-dev/pixi#742] [Ideas]
Hey all, 

Pixi is fast :fire: , but its heavily limited by the downloading speed of your internet. This is a problem that is not easy to solve. But while you're waiting, the feedback of what pixi is doing could be better, we know!

I want to start a discussion to brainstorm ideas for **better user feedback** in the open! Everyone can chip in!

## What would I want?
- An ordered output per environment, per platform, per package. When downloading/solving/installing, I want to know precisely from which package the bottleneck is coming.
- I want the progress to be easy on the eyes, we don't need big disco lights, but it can help if certain types of information are displayed in the same styling, e.g. the environment name is always bold an blue, a platform is always dim and black on white/visa versa.
- At the end of an installation I would like some information, anywhere. Currently almost all information is removed and a green check is shown that the environment is ready. 
  - When running `pixi install` I would like a little more info about the resulting locked packages, and the environment that is installed.
  - When running `pixi run/shell` it should just continue, here I don't need installation information unless i'm asking for it. 
- I would like to see what is done already until it is completely done, now everything that is done gets removed immediately from the terminal. 

## Some pseudo ideas
Using normal `indicatif` terminal process bars could look something like:
```
~/projectx on -> main  
➜ pixi install
⣯ Getting `projectx` ready 
  ⣟ Environment `default`
    ✔ Solving environment
    	✔ linux-64
        ✔ osx-64
    ⣯ Downloading packages 26/160
         tensorflow 0.1.2 [━━━╾─────]   16.8 MiB/s
         cudnn      12.8  [━━━━━━━╾─]   1.6 MiB/s
  ⣟ Environment `lint`
      ✔ Solving environment
          ✔ linux-64
          ✔ osx-64
      ⣯ Downloading packages 0/1
         pre-commit 0.1.2 [━━━╾─────]   0.8 MiB/s
  ⣟ Environment `pypi`
      ✔ Solving environment
        ✔ linux-64
        ⣯ osx-64
	  ⣯ Resovling dependencies 6/18
	      ⣯ Building wheel for: sdist 0.0.0
  ✔ Environment `tiny`
      ✔ Solving environment
        ✔ linux-64
        ✔ osx-64
      ✔ Downloaded/installed packages 10/10  
```
But as you see even for a small example it already can look very intents. 

That is why it would be even more awesome to have another solution with an more advanced UI tool like [`ratatui`](https://ratatui.rs/showcase/apps/) Which can support a scroll-able/collapsible output while its solving, with the possibility to keep it open and use it to view your progress. 

We're looking for good ideas, and possibly prototypes of the solutions above!

**Comment by lucascolley** [comment]
I think there has been a lot of improvement here :)
